pandas/tests/extension/arrow/test_arrow.py

```diff
@@ -0,0 +1,185 @@
+from datetime import (
+    date,
```

Comment:
I would maybe move this file one level up for now (directly as `test_arrow.py` inside the `extension` directory). That would follow the pattern for all our other built-in extension dtypes (those subdirectories like `arrow` (or json, decimal, etc) are all tests with a custom EA implemented specifically for the tests). 

And if we do that, can you add the same disclaimer as we have in all the other test files for builtin EAs, like https://github.com/pandas-dev/pandas/blob/3bf2cb1b227c80461c7a736718ae17e35d6d5772/pandas/tests/extension/test_boolean.py#L1-L14

Reply:
Will do.

Side note, will we still need the tests in `pandas/tests/extension/arrow` once we get ArrowExtensionArray fully supported?

Reply:
> Side note, will we still need the tests in pandas/tests/extension/arrow once we get ArrowExtensionArray fully supported?

We would need to check, they might be testing very specific cases that came up in the past that are not covered by the main base extension tests (but probably they could be re-written to use the built-in ArrowExtensionArray)